server/apiv2/handlers/stores.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/tikv/pd/pkg/apiutil"
+	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/server/cluster"
+)
+
+// GetStores returns all stores.
+// @Tags stores
+// @version 2.0
+// @Summary Get all stores' information.
+// @Param id path integer true "Store Id"
+// @Produce json
+// @Success 200 {object} StoresInfo
+// @Failure 404 {string} string "The store does not exist."
+// @Router /stores [get]
+func GetStores() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		rc := c.MustGet(apiutil.ClusterKey).(*cluster.RaftCluster)
+		stores := rc.GetMetaStores()
+		storesInfo := &StoresInfo{
+			Stores: make([]*StoreInfo, 0, len(stores)),
+		}
+
+		nodeStates, exist := c.GetQueryArray("node_state")
+		for _, s := range stores {
+			storeID := s.GetId()
+			store := rc.GetStore(storeID)
+			if store == nil {
+				c.AbortWithStatus(http.StatusNotFound)
```

Comment:
Should we just ignore it? since there is no lock, so I think it is possible that a store maybe removed after `rc.GetMetaStores`